test/e2e/checkpoint_test.go

```diff
@@ -933,18 +935,28 @@ var _ = Describe("Podman checkpoint", func() {
 	})
 
 	It("podman checkpoint and restore container with different port mappings", func() {
-		localRunString := getRunString([]string{"-p", "1234:6379", "--rm", redis})
+		rand.Seed(time.Now().UnixNano())
+		orginalPort := 37000 + rand.Intn(2000) + GinkgoParallelProcess()
+		localRunString := getRunString([]string{"-p", fmt.Sprintf("%d:6379", orginalPort), "--rm", redis})
```

Comment:
@adrianreber this may be my ignorance, but I wonder if you could [use `WaitContainerReady()` here](https://github.com/containers/podman/blob/main/test/utils/utils.go#L179) instead of blindly entering the connect/timeout loop?

Reply:
I just tried it and it also solves the problem locally for me. I will update this PR.